packages/cli/src/models/local/KitController.ts

```diff
@@ -0,0 +1,116 @@
+import child from '../../utils/child';
+import Spinner from '../../utils/spinner';
+
+const simpleGit = patch('simple-git/promise');
+
+export const DEFAULT_VERSION = '0.1.0';
+
+export interface KitConfig {
+  version: string;
+  message: string;
+  files: string[];
+  hooks: object;
+}
+
+/* tslint:disable */
+// schema has to have doulbe quotes
+const kitConfigSchema = {
+  "type": "object",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "properties": {
+    "version": { "type": "string" },
+    "message": { "type": "string" },
+    "files": { "type": "array" },
+    "hooks": { "type": "object" },
+  },
+  "additionalProperties": false,
+  "required": ["version", "message", "files", "hooks"],
+};
+/* tslint:enable */
+
+export default class KitController {
+  public async unpack(url: string, workingDirPath: string = '', config: KitConfig): Promise<void | never> {
+    if (!url) throw Error('A url must be provided.');
+    if (!config) throw Error('A config must be provided.');
+
+    const { exec } = child;
+    const { readdir, remove, pathExists } = fs;
+
+    // because zos always spawns '.zos.lock' file
+    if ((await readdir(workingDirPath)).length > 1) throw Error('The directory must be empty');
+
+    let spinner = new Spinner(`Downloading kit from ${url}`);
+    try {
+      spinner.start();
+      const git = simpleGit(workingDirPath);
+      await git.init();
+      await git.addRemote('origin', url);
+      await git.fetch();
+      // if files are empty checkout everything
+      if(!config.files.length) {
+        await git.pull('origin', 'stable');
+      } else {
+        // if there are some files then do tree-ish checkout
+        // http://nicolasgallagher.com/git-checkout-specific-files-from-another-branch/
+        await git.checkout([
+          `origin/stable`,
+          `--`,
+          ...config.files,
+        ]);
+      }
+      spinner.succeed();
+
+      spinner = new Spinner('Unpacking kit');
+      spinner.start();
+      // always delete .git folder
+      await remove(path.join(workingDirPath, '.git'));
+      // run kit commands like `npm install`
+      await exec(config.hooks['post-unpack']);
+      spinner.succeed();
+
+      stdout('The kit is ready to use. Amazing!');
+      stdout(config.message);
+    } catch(e) {
+      spinner.fail();
+      // TODO: remove all files from directory on fail except .zos.lock
+      e.message = `Failed to download and unpack kit from ${url}. Details: ${e.message}`;
+      throw e;
+    }
+  }
+
+  public async verifyRepo(url: string): Promise<KitConfig | never> {
+    if (!url) throw Error('A url must be provided.');
+
+    try {
+      const config = (await axios.get(url
+        .replace('.git', '/stable/kit.json')
+        .replace('github.com', 'raw.githubusercontent.com')
+      )).data;
+
+      // validate our json config
+      // TODO: derive the schema directly from the KitConfig type
+      const ajv = new Ajv({ allErrors: true });
```

Comment:
I wasn't familiar with `ajv`, very cool! We should start using more throughout the app.